src/orange/OrangeParams.cc

```diff
@@ -114,15 +97,34 @@ OrangeParams::OrangeParams(const std::string& json_filename)
  * Volume and surface labels must be unique for the time being.
  */
 OrangeParams::OrangeParams(OrangeInput input)
-    : unit_indexer_(make_unit_indexer(input.units))
 {
     CELER_VALIDATE(!input.units.empty(), << "input geometry has no universes");
 
-    // Insert all units
     HostVal<OrangeParamsData> host_data;
-    detail::UnitInserter      insert_unit(&host_data);
+
+    // Calculate offsets for UnitIndexerData
+    make_builder(&host_data.unit_indexer_data.surfaces).push_back(0);
+    make_builder(&host_data.unit_indexer_data.volumes).push_back(0);
+    for (const UnitInput& u : input.units)
+    {
+        using AllVals = AllItems<size_type, MemSpace::native>;
+
+        auto surface_offset
+            = host_data.unit_indexer_data.surfaces[AllVals{}].back();
+        auto volume_offset
+            = host_data.unit_indexer_data.volumes[AllVals{}].back();
+
+        make_builder(&host_data.unit_indexer_data.surfaces)
+            .push_back(surface_offset + u.surfaces.size());
+        make_builder(&host_data.unit_indexer_data.volumes)
+            .push_back(volume_offset + u.volumes.size());
```

Comment:
When I wrote the "collection builder" helper class I expected it to be created locally and reused, but in practice we rarely do that because of scoping issues. Here though would be a good opportunity:
```suggestion
    // Calculate offsets for UnitIndexerData
    auto ui_surf = make_builder(&host_data.unit_indexer_data.surfaces);
    auto ui_vol = make_builder(&host_data.unit_indexer_data.volumes);
    unit_indexer_surfaces.push_back(0);
    unit_indexer_volumes.push_back(0);
    for (const UnitInput& u : input.units)
    {
        using AllVals = AllItems<size_type, MemSpace::native>;

        auto surface_offset
            = host_data.unit_indexer_data.surfaces[AllVals{}].back();
        auto volume_offset
            = host_data.unit_indexer_data.volumes[AllVals{}].back();

        ui_surf.push_back(surface_offset + u.surfaces.size());
        ui_vol.push_back(volume_offset + u.volumes.size());
```